### PR TITLE
Ritika| Week5 - Redux counter application

### PR DESCRIPTION
**Pull request by RitikaGarg-ITT**
Pull Request Questionnaire

(1) Which Jira ticket(s) is this pull request for, and is the ticket now complete?

(2) Briefly describe your change.
I have integrated my week-5 assignment in this joint project.  I have added redux tool kit to make this counter application with increment decrement and increment by Value features.

(3) What parts of the application are affected by your change?
It is affecting Ritika's week-5 component.

(4) How did you test your change? If relevant, which role(s) and/or customer configuration(s) did you use?
Tested my application on various edge cases and some basic scenarios

Root Cause (if any)?
-

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -41,12 +42,13 @@ const App: React.FC = () => {
             <Route path='/aditya/week3' element={<AdityaWeek3 />}></Route>
           </Route>
           <Route path='ritika' element={<Ritika />}>
-            <Route path='week2' element={<Week2 />}></Route>
+            <Route path='week2' element={<Week2 />} />
             <Route path='week3' element={<Week3Main />}></Route>
             <Route path='week4' element={<Week4Main />}>
               <Route path='home' element={<Week4Home />}></Route>
               <Route path='movie/:id' element={<MovieDisplay />}></Route>
```
Comment (aditya-s-19) [inline src/App.tsx]: use self closing tags in these

File: src/components/Home/Ritika/Week5/Counter/counter.styled.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
+import styled from 'styled-components';
+
+export const CounterWrapper = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  background-color: #f5f5f5;
+  height: 100vh;
+
```
Comment (aditya-s-19) [inline src/components/Home/Ritika/Week5/Counter/counter.styled.ts]: remove unnecessary line spaces

File: src/components/Home/Ritika/Week5/Counter/counter.styled.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    padding: 1%;
+    border: 1px solid #cccccc;
+    border-radius: 5%;
+    font-size: 1.5vw;
+    margin-right: 2%;
+    border: none;
+    outline: none;
+    width: 150px;
+    box-shadow:
+      rgba(50, 50, 93, 0.25) 0px 50px 100px -20px,
```
Comment (aditya-s-19) [inline src/components/Home/Ritika/Week5/Counter/counter.styled.ts]: use hex value here also for consistency

File: src/components/Home/Ritika/Week5/Counter/Counter.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  return (
+    <CounterWrapper>
+      <button className='counterButton' onClick={() => dispatch(increment())}>
+        +
+      </button>
+      <div className='counterValue'>{value}</div>
+      <button className='counterButton' onClick={() => dispatch(decrement())}>
+        -
+      </button>
+      <input className='incrementByValueInput' type='text' value={inputValue} onChange={handleInputChange}></input>
```
Comment (aditya-s-19) [inline src/components/Home/Ritika/Week5/Counter/Counter.tsx]: use self closing tag here also

**Review by aditya-s-19 — DISMISSED** [review]

**Review by anusreesubash — DISMISSED** [review]

File: src/components/Home/Ritika/Week3/utils/constants.json [hunk 5]
```diff
@@ -3,4 +3,5 @@
     "autoFocusInputWarning":"Please provide both name and phone number.",
     "alreadyExistUser":"Employee with this name already exists!",
     "successMessage":"Employee added successfully"
+
```
Comment (Harrrrry) [inline src/components/Home/Ritika/Week3/utils/constants.json:7]: Why this extra line here?

File: src/components/Home/Ritika/Week2/utils/constants/constants.json [hunk 6]
```diff
@@ -1,6 +1,7 @@
 {
     "properDetailsWarning": "Please fill the details properly!!!",
     "validDateWarning": "Please select a valid date for completion!!",
-    "deleteConfirmation":" Are you sure you want to delete the task?"
-    
+    "deleteConfirmation":" Are you sure you want to delete the task?",
+    "passwordRequirements":"Password must contain at least 8 characters, including uppercase, lowercase, number, and a special character.",
+    "successLoggedIn":"Logged In Successfully"
```
Comment (Harrrrry) [inline src/components/Home/Ritika/Week2/utils/constants/constants.json]: Why all word start with caps for this message?

File: src/components/Home/Ritika/Login/Login.tsx [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+  const handleUserNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setUserName(e.target.value);
+  };
+  const handleUserPasswordChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setUserPassword(e.target.value);
+  };
+
+  const handleSubmit = () => {
+    if (!validateUser(userName, userPassword)) return;
```
Comment (Harrrrry) [inline src/components/Home/Ritika/Login/Login.tsx]: Here, I think user validation is only required here no required to password validation because you are doing login.
Password validation is required at the time of signup

File: src/components/Home/Ritika/Week3/utils/validation.ts [hunk 8]
```diff
@@ -23,3 +23,14 @@ export const validateAlreadyExistUser = (employees: Employee[], name: string) =>
   }
   return true;
 };
+export const validateUser = (userName: string, userPassword: string) => {
+  if (!userName.trim() || !userPassword.trim()) {
+    toast.error('Please enter the details');
```
Comment (Harrrrry) [inline src/components/Home/Ritika/Week3/utils/validation.ts]: Can you put this message as well in constant file

File: src/components/Home/Ritika/Week3/utils/constants.json (hunk 5 above)
Comment (RitikaGarg-ITT) [reply src/components/Home/Ritika/Week3/utils/constants.json:7]: this is the eol, an eslint feature

Comment (shreegovinda) [reply src/components/Home/Ritika/Week3/utils/constants.json:7]: eol should be lastline (ie after line 8) not before here

**Review by shreegovinda — DISMISSED** [review]

**Review by shreyansh-khatri — DISMISSED** [review]

File: src/components/Home/Ritika/Week3/utils/constants.json (hunk 5 above)
Comment (RitikaGarg-ITT) [reply src/components/Home/Ritika/Week3/utils/constants.json:7]: Oh okay , my mistake shree, may be I did the changes with an extra line and pushed the code. But I had resolved this comment as well

**Review by aditya-s-19 — APPROVED** [review]

**Review by shreyansh-khatri — APPROVED** [review]
